config/src/main/java/org/springframework/security/config/annotation/web/builders/WebSecurity.java

```diff
@@ -208,6 +205,25 @@ public WebSecurity addSecurityFilterChainBuilder(
 		return this;
 	}
 
+	/**
+	 * <p>
+	 * Adds {@link SecurityFilterChain} instances.
+	 * </p>
+	 *
+	 * <p>
+	 * Typically this method is invoked automatically within the framework from
+	 * {@link WebSecurityConfigurerAdapter#init(WebSecurity)}
+	 * </p>
+	 *
+	 * @since 5.4
+	 * @param securityFilterChain the {@link SecurityFilterChain} instance
+	 * @return the {@link WebSecurity} for further customizations
+	 */
+	public WebSecurity addSecurityFilterChain(SecurityFilterChain securityFilterChain) {
+		this.securityFilterChains.add(securityFilterChain);
```

Comment:
I wonder if we could/should use `this.securityFilterChainBuilders.add(() -> securityFilterChain)` instead. This would mean that performBuild could stay the same and be simplified (it only operates on a single collection instead of two.